docs/dev/index.rst

```diff
@@ -15,28 +15,327 @@
     specific language governing permissions and limitations
     under the License.
 
-Design and Developer Guide
-==========================
+Design and Architecture
+=======================
+
+This document is intended for developers who want to understand the
+architecture of TVM and/or actively develop on the project.
+This page is organized as follows:
+
+- The `Example Compilation Flow Walkthrough`_ section is a walk through of a typical compilation flow explaining each component used during compilation.
```

Comment:
'Example Compilation Flow' a better name for this section.

A better description could be:
The Example Compilation Flow section gives an overview of the steps that TVM takes to turn a high level description of a model into an optimized function.